internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumer.java

```diff
@@ -0,0 +1,63 @@
+package com.linkedin.venice.client.consumer;
+
+import com.linkedin.venice.annotation.Experimental;
+import com.linkedin.venice.pubsub.api.PubSubMessage;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+
+/**
+ * Venice change capture consumer to provide
+ * @param <K>
+ * @param <V>
+ */
+@Experimental
+public interface VeniceChangelogConsumer<K, V> {
+  /**
+   * Subscribe a set of partitions to this VeniceChangelogConsumer. The VeniceChangelogConsumer
+   * should try to consume messages from all partitions that are subscribed to it.
+   *
+   * @param partitions the set of partition to subscribe and consume
+   * @return a future which completes when the partitions are ready to be consumed data
+   * @throws a VeniceException if subscribe operation failed for any of the partitions
+   */
+  CompletableFuture<Void> subscribe(Set<Integer> partitions);
+
+  /**
+   * Subscribe all partitions belonging to a specific store.
+   *
+   * @return a future which completes when all partitions are ready to be consumed data
+   * @throws a VeniceException if subscribe operation failed for any of the partitions
+   */
+  CompletableFuture<Void> subscribeAll();
+
+  /**
+   * Stop ingesting messages from a set of partitions for this VeniceChangelogConsumer.
+   *
+   * @param partitions The set of topic partitions to unsubscribe
+   * @throws a VeniceException if unsubscribe operation failed for any of the partitions
+   */
+  void unsubscribe(Set<Integer> partitions);
+
+  /**
+   * Stop ingesting messages from all partitions.
+   *
+   * @throws a VeniceException if unsubscribe operation failed for any of the partitions
+   */
+  void unsubscribeAll();
+
+  /**
+   * Polling function to get any available messages from the underlying system for all partitions subscribed.
+   *
+   * @param timeoutInMs The maximum time to block (must not be greater than {@link Long#MAX_VALUE} milliseconds)
```

Comment:
add to block/wait in between two polling requests?

Reply:
I think block indication for this `poll` function is enough, caller should be aware of this behavior.

Reply:
We just need to make it explicit in the doc string